vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -402,12 +441,40 @@ void AdsImpl::NotificationEventDismissed(
 void AdsImpl::NotificationEventTimedOut(
     const std::string& id,
     const NotificationInfo& notification) {
-  notifications_->Remove(id);
+  notifications_->Remove(id, false);
 
   GenerateAdReportingNotificationResultEvent(notification,
       NotificationResultInfoResultType::TIMEOUT);
 }
 
+bool AdsImpl::IsDoNotDisturb() const {
+  if (!IsAndroid()) {
+    return false;
+  }
+
+  auto now = base::Time::Now();
+  base::Time::Exploded now_exploded;
+  now.LocalExplode(&now_exploded);
+
+  if (now_exploded.hour >= kDoNotDisturbToHour &&
+      now_exploded.hour <= kDoNotDisturbFromHour) {
+    return false;
+  }
+
+  return true;
+}
+
+bool AdsImpl::IsAndroid() const {
```

Comment:
why are we creating a custom method here for an android check when one already exists `defined(OS_ANDROID)` ?

Reply:
With different business logic across mobile and desktop it speeds up the development of features, as then we only need to run a final sanity check on the specific platforms at the end. i.e. I test ad delivery, catalog logic etc. initially on a single device

Reply:
for example see `if (!IsAndroid() && !IsForeground()) {` below

Reply:
Issues raised https://github.com/brave/brave-browser/issues/6229